src/js/api/api.js

```diff
@@ -42,36 +43,42 @@ define([
         var _this = this,
             _instream,
             _originalContainer = container,
-            _controller = new Controller(),
+            _controller,
             _playerReady = false,
-            _itemMeta = {},
-            _callbacks = {};
+            _itemMeta = {};
 
         // Set up event handling
         _.extend(this, Events);
 
 
         this.trigger = function(type, args) {
-            args = (_.isObject(args) ? args : {});
+            if (_.isObject(args)) {
+                args = normalizeOutput(args);
+            } else {
+                args = {};
+            }
 
-            args.type = args.type || type;
-            args = normalizeOutput(args);
+            // map events
+            if (type === 'ready') {
+                _playerReady = true;
+                _qoe.tick('ready');
+                args.setupTime = _qoe.between('setup', 'ready');
+            } else if (type === events.JWPLAYER_PLAYER_STATE) {
+                type = args.newstate;
+            }
```

Comment:
I don't think we should add this logic into the trigger function. An alternative solution to the problem of triggering "ready" events is to move the logic from API into controller. I put an example here https://github.com/jwplayer/jwplayer/commit/c913b5c9a5605244fd6bcdd62c31c849bb0cc488